### PR TITLE
fix(web): tracking events in studio

### PR DESCRIPTION
**Pull request by davidsoderberg**

### CODE REVIEW

File: apps/web/src/api/api.client.ts [hunk 1]
```diff
@@ -128,6 +128,17 @@ export function buildApiHttpClient({
     }
   };
 
+  const post = async (url, body?: Record<string, unknown>) => {
```
Comment (antonjoel82) [inline apps/web/src/api/api.client.ts]:
```suggestion
  async function post<TBody = Record<string, unknown>> (url: string, body?: TBody) {
```

File: apps/web/src/utils/segment.ts [hunk 2]
```diff
@@ -1,14 +1,26 @@
...
+
+type TrackFunction = UseMutateFunction<
+  any,
+  unknown,
+  {
+    event: string;
+    data?: Record<string, unknown> | undefined;
+  },
+  unknown
+>;
```
Comment (antonjoel82) [inline apps/web/src/utils/segment.ts]: 💭 **thought (non-blocking):** We should type this more intentionally‏

File: apps/web/src/components/providers/SegmentProvider.tsx [hunk 3]
```diff
@@ -9,7 +10,8 @@ type Props = {
 const SegmentContext = React.createContext<SegmentService>(undefined as any);
 
 export const SegmentProvider = ({ children }: Props) => {
-  const segment = React.useMemo(() => new SegmentService(), []);
+  const { mutate: track } = useTelemetry();
```
Comment (rifont) [inline apps/web/src/components/providers/SegmentProvider.tsx:13]: ❓ **question:** Can we modify `useTelemetry` so we can do `const { track } = useTelemetry();` instead? This is a common pattern and object destructure renaming will be cumbersome + bad inner-DX‏

**Review by rifont — APPROVED** [review]
Nice work! 🚀

File: apps/web/src/utils/segment.ts [hunk 4]
```diff
@@ -1,14 +1,26 @@
...
 
 export class SegmentService {
   private _segment: AnalyticsBrowser | null = null;
+  private readonly _trackFunction: TrackFunction;
   private _segmentEnabled: boolean;
   public _mixpanelEnabled: boolean;
 
-  constructor() {
+  constructor(trackFunction: TrackFunction) {
```
Comment (antonjoel82) [inline apps/web/src/utils/segment.ts]: ✏ **suggestion:** Convert this to an object to make it easier to extend with a smaller changeset in the future‏

**Review by scopsy — APPROVED** [review]

File: apps/web/src/hooks/useNovuAPI.ts [hunk 5]
```diff
@@ -33,3 +34,28 @@ export const useApiKeys = (options?: any) => {
...
+      mixpanel.register({ $device_id: segmentDeviceId });
+      const sessionReplayProperties = mixpanel.get_session_recording_properties();
+
+      data = {
+        ...(data || {}),
+        ...sessionReplayProperties,
+      };
+    }
+
+    return mutate({ event: event + ' - [WEB]', data });
```
Comment (SokratisVidros) [inline apps/web/src/hooks/useNovuAPI.ts:59]:
```suggestion
    return mutate({ event: event + ' - [Dashboard]', data });
```

**Review by SokratisVidros — APPROVED** [review]

**Pull request merged**